crypto.go

```diff
@@ -82,14 +82,15 @@ func (rd *RedisStorage) DecryptStorageData(bytes []byte) (*StorageData, error) {
 		return nil, err
 	}
 
-	// Simple sanity check of the beginning of the byte array just to check
-	if len(bytes) < len(rd.ValuePrefix) || string(bytes[:len(rd.ValuePrefix)]) != rd.ValuePrefix {
-		return nil, fmt.Errorf("invalid data format")
+	// Strip unused prefix from the beginning of the byte array if present
+	var PrefixLength = len(rd.ValuePrefix)
+	if len(bytes) >= PrefixLength && string(bytes[:PrefixLength]) == rd.ValuePrefix {
```

Comment:
You can use `strings.HasPrefix()` and `strings.TrimPrefix()` btw https://pkg.go.dev/strings#TrimPrefix

Reply:
Thanks for the suggestion @francislavoie using those string functions would be more elegant however I believe that would require the `bytes[]` byte array to be converted to a string first? I'm not sure if that going to be any more efficient in this case.